### DISCUSSION TITLE
2024-10-23-r1

### DISCUSSION THREAD

**Discussion by Zoey2936** [ZoeyVid/NPMplus#1185] [Announcements]
# Breaking changes in last release, please read the last changelog here: https://github.com/ZoeyVid/NPMplus/releases/tag/2024-10-21-r1

## What's Changed
- doc updates
- dep updates
- merge https://github.com/ZoeyVid/NPMplus/pull/1182 by @Sproglet (keep goaccess data after rotating logs)
- run ocsp script before testing nginx config
- fix updates from older versions
- remove useless headers
- move some files
- change chipper order

### How to update
- Read the changes above
- Pull the zoeyvid/npmplus:latest image
- apply possible changes that maybe effect you from above to your compose.yaml/NPMplus
- redeploy the compose stack
- report any issues you find

**Full Changelog**: https://github.com/ZoeyVid/NPMplus/compare/2024-10-22-r1...2024-10-23-r1

<hr /><em>This discussion was created from the release <a href='https://github.com/ZoeyVid/NPMplus/releases/tag/2024-10-23-r1'>2024-10-23-r1</a>.</em>

**Comment by gadget78** [comment]
if you "do not" set the ACME_EMAIL in the compose.yml file (and instead you # it out)
then when 1st run you get a error about reading the certbot.ini file... 

```
-------------------------------------
 _ _  ___  __ __       _
| \ || . \|  \  \ ___ | | _ _  ___
|   ||  _/|     || . \| || | |[_-[
|_\_||_|  |_|_|_||  _/|_| \__|/__/
                 |_|
-------------------------------------
Version:  2.12.1+5b65de0
Date:     Thu Oct 24 20:57:59 BST 2024
User:     root
PUID:     0
User ID:  0
PGID:     0
Group ID: 0
-------------------------------------

usage: 
  certbot [SUBCOMMAND] [options] [-d DOMAIN] [-d DOMAIN] ...

Certbot can obtain and install HTTPS/TLS/SSL certificates.  By default,
it will attempt to use a webserver both for obtaining and installing the
certificate. 
certbot: error: Unable to open config file: /etc/tls/certbot.ini. Error: No such file or directory
```

if i alter the launch.sh file (lines 56, 61, and 66) for eg...
https://github.com/ZoeyVid/NPMplus/blob/7a2799b5f28389ef585cc0e2c05657b59aae89c1/rootfs/usr/local/bin/launch.sh#L56
and edit `--config /etc/tls/certbot.ini` to `--config /etc/certbot.ini`

it then works :)

**Reply by Zoey2936** [reply]
will be fixed

**Reply by gadget78** [reply]
excellent, will give that a try... 
ive also found another issue, althou i "think" it may of been intentional, but im not able to find the "fix"/"new method"

when using the API to create a access token, i use "expiry", to create a token i can then use longer than 24hours...
but when trying to use expiry within the token request i get 
`{"error":{"code":400,"message":"data must NOT have additional properties"}}`

so when i use 
`curl -k -s -m 10 "https://<IP>:81/api/tokens" -H "Content-Type: application/json; charset=UTF-8" --data-raw "{\"identity\":\"<EMAIL>\",\"secret\":\"<PASSWORD>\",\"expiry\":\"1y\"}"`
is there a new endpoint ? or another way to add expiry ?
i see within token.js does still have expiry functions in `getFreshToken` but im not sure how we can access this via the tokens API endpoint ? ..

**Reply by Zoey2936** [reply]
This because of an upstream change:
how it was before:
https://github.com/NginxProxyManager/nginx-proxy-manager/blob/v2.11.3/backend/schema/endpoints/tokens.json
and after:
https://github.com/ZoeyVid/NPMplus/blob/develop/backend/schema/paths/tokens/post.json
as you see expiry was removed

**Reply by Zoey2936** [reply]
But you could try to change additionalProperties to true in /app/schema/paths/tokens/post.json (inside the container), then restart npmplus (not recreate) and then tets it and report back

**Reply by gadget78** [reply]
yep, changing additionalProperties to true in /app/schema/paths/tokens/post.json DOES get it working :)

**Reply by Zoey2936** [reply]
should be fixed in develop
